### ISSUE TITLE
Bad out-of-box support for PDM-based builds

### ISSUE THREAD

**Issue by NickCrews** [airspeed-velocity/asv#1356]
I am adding ASV to [mismo](https://github.com/nickcrews/mismo/), which uses PDM as the project manager and build tool.

With no `build_command` specified in the asv.conf.json, then since the project has a pyproject.toml, the following default build command is used:

https://github.com/airspeed-velocity/asv/blob/b8cbf81ccb808158102692a6fc42b8a9b2f2861e/asv/environment.py#L918-L923

However, this fails, because the build dependency of `pdm.backend`, but the various flags there disallow going to PyPI to fetch it (as I understand it, perhaps the root cause is different). In order to get the build to work, I have to use
`python -mpip wheel -w {build_cache_dir} {build_dir}`

Why do we disallow fetching to PyPI? It seems like fetching the build backend is going to be needed in many cases, including pdm.backend, poetry-core, and hatchling? Is there something else we can do to make the out-of-box experience better for these users?

**Comment by HaoZeke** [comment]
Thanks for the issue. I believe the idea is to be able to run without network connectivity. How reasonable that is today in 2024 is an open question, but much of the design internals explicitly disallow outbound network calls..

**Comment by HaoZeke** [comment]
This sounds like a pragmatic solution for `pdm` and most of the build backends, would you be open to adding this to the documentation?

**Comment by NickCrews** [comment]
What would happen if we just changed the default build command to what I suggested? It seems like that would strictly increase the number of environments that would work out of the box? Would that swap somehow break people's existing environments that don't have network? It seems like they would maybe just fail in a different place?

**Comment by NickCrews** [comment]
My thought would be that we should assume we have network connectivity, I think that will help more people than it will hurt.

If we don't actually change the code then I can change the documentation

**Comment by HaoZeke** [comment]
> My thought would be that we should assume we have network connectivity, I think that will help more people than it will hurt.

I agree. Worth a shot. A few releases ago ASV used to have build isolation on as well, the progressive weakening is fine (will likely hear quickly enough from the users if things change). Could you make a PR for this? I'm planning to cut a new bugfix release early this week (or even today if everything extant is in).
